pkg/networkservice/chains/nsmgr/server.go

```diff
@@ -69,20 +70,76 @@ type nsmgrServer struct {
 	registry.Registry
 }
 
+type serverOptions struct {
+	authorizeServer networkservice.NetworkServiceServer
+	dialOptions     []grpc.DialOption
+	registryCC      grpc.ClientConnInterface
+	name            string
+	url             string
+}
+
+// Option modifies server option value
+type Option func(o *serverOptions)
+
+// WithDialOptions sets gRPC Dial Options for the server
+func WithDialOptions(dialOptions ...grpc.DialOption) Option {
+	return func(o *serverOptions) {
+		o.dialOptions = dialOptions
+	}
+}
+
+// WithAuthorizeServer sets authorization server chain element
+func WithAuthorizeServer(authorizeServer networkservice.NetworkServiceServer) Option {
+	if authorizeServer == nil {
+		panic("Authorize server cannot be nil")
+	}
+	return func(o *serverOptions) {
+		o.authorizeServer = authorizeServer
+	}
+}
+
+// WithRegistryClientConn sets client connection to reach the upstream registry, could be nil, in this case only in memory storage will be used.
+func WithRegistryClientConn(registryCC grpc.ClientConnInterface) Option {
+	return func(o *serverOptions) {
+		o.registryCC = registryCC
+	}
+}
+
+// WithName - set a nsmgr name, a default name is `Nsmgr`.
+func WithName(name string) Option {
+	return func(o *serverOptions) {
+		o.name = name
+	}
+}
+
+// WithURL - set a public URL address for NetworkServiceManager, it is used to access endpoints within this NSMgr from remote clients.
+// Default value is not set, and endpoints address will not be changed.
+func WithURL(url string) Option {
```

Comment:
URL should be required parameter.

Reply:
It is not required for Local -> Nsmgr -> local only cases and for cases if we doesn't use public Url for nsmgr.

Reply:
OK, currently we have not a test for this scenario actually. Let's then cover it :)

Reply:
Added test with a Unix Socket nsmgr and empty Url.